tracer/src/Datadog.Trace/Agent/StatsAggregator.cs

```diff
@@ -105,6 +113,30 @@ public void AddRange(Span[] spans, int offset, int count)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public ArraySegment<Span> ProcessTrace(ArraySegment<Span> trace)
+        {
+            if (CanComputeStats == true || CanComputeStats is null)
```

Comment:
Super nit - I think it's a comment for the previous PR, but maybe we should not use `CanComputeStats` directly and hide the complexity of true or null behind a helper.

Reply:
I'm thinking of keeping this here for now, mostly because I have the follow-up PR which could turn off the stats if the agent is found not to be compatible. We can come back to this if it becomes a headache